controller.rst

```diff
@@ -336,6 +336,157 @@ object. To access it in your controller, add it as an argument and
 :ref:`Keep reading <request-object-info>` for more information about using the
 Request object.
 
+.. _controller_map-request:
+
+Automatic Mapping Of The Request
+--------------------------------
+
+It is possible to automatically map request's payload and/or query parameters to
+your controller's action arguments with attributes.
+
+Mapping Query Parameters Individually
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say a user sends you a request with the following query string:
+``https://example.com/dashboard?firstName=John&lastName=Smith&age=27``.
+Thanks to the :class:`Symfony\\Component\\HttpKernel\\Attribute\\MapQueryParameter`
+attribute, arguments of your controller's action can be automatically fulfilled::
+
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\HttpKernel\Attribute\MapQueryParameter;
+
+    // ...
+
+    public function dashboard(
+        #[MapQueryParameter] string $firstName,
+        #[MapQueryParameter] string $lastName,
+        #[MapQueryParameter] int $age,
+    ): Response
+    {
+        // ...
+    }
+
+``#[MapQueryParameter]`` can take an optional argument called ``filter``. You can use the
+`Validate Filters`_ constants defined in PHP::
+
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\HttpKernel\Attribute\MapQueryParameter;
+
+    // ...
+
+    public function dashboard(
+        #[MapQueryParameter(filter: \FILTER_VALIDATE_REGEXP, options: ['regexp' => '/^\w++$/'])] string $firstName,
+        #[MapQueryParameter] string $lastName,
+        #[MapQueryParameter(filter: \FILTER_VALIDATE_INT)] int $age,
+    ): Response
+    {
+        // ...
+    }
+
+.. versionadded:: 6.3
+
+    The :class:`Symfony\\Component\\HttpKernel\\Attribute\\MapQueryParameter` attribute
+    was introduced in Symfony 6.3.
+
+Mapping The Whole Query String
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Another possibility is to map the entire query string into an object that will hold
+available query parameters. Let's say you declare the following DTO with its
+validation constraints::
```

Comment:
```suggestion
optional validation constraints::
```

to decouple the fact that we can map without any constraint rules :)

Reply:
Good suggestion in my opinion, added! :slightly_smiling_face: